### PR TITLE
[KOGITO-1768] allow code scaffolding in kogito-maven-plugin

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

File: kogito-quarkus-extension/deployment/src/main/java/org/kie/kogito/quarkus/deployment/KogitoAssetsProcessor.java [hunk 1]
```diff
@@ -69,7 +68,9 @@
 
 public class KogitoAssetsProcessor {
 
-    private static final String generatedDashboardsDir = "/target/resources/";
+    private static final String generatedResourcesDir = System.getProperty("kogito.codegen.sources.directory", "target/generated-resources/kogito");
```
Comment (evacchi) [inline kogito-quarkus-extension/deployment/src/main/java/org/kie/kogito/quarkus/deployment/KogitoAssetsProcessor.java:71]: I think we should use quarkus config for this

File: kogito-quarkus-extension/deployment/src/main/java/org/kie/kogito/quarkus/deployment/KogitoAssetsProcessor.java [hunk 2]
```diff
@@ -239,31 +233,46 @@ public void generateModel(ArchiveRootBuildItem root,
...
             } catch (IOException e) {
-                logger.warn(String.format("Could not write resource file %s", f.toString()), e);
+                logger.warn(String.format("Could not write file '%s'", f.toString()), e);
             }
-        });
+        }
     }
 
+    private boolean isCustomizable(GeneratedFile f) {
+        return f.relativePath().endsWith("Resource.java")
```
Comment (evacchi) [inline kogito-quarkus-extension/deployment/src/main/java/org/kie/kogito/quarkus/deployment/KogitoAssetsProcessor.java:257]: I recall doing this change myself for the PoC, let's not do this in "real" code :D

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java [hunk 3]
```diff
@@ -178,6 +178,7 @@ private static ResourceType typeOf(FileSystemResource r) {
     private KieModuleModel kieModuleModel;
     private boolean hotReloadMode = false;
     private boolean useMonitoring = false;
+    private boolean useWebServices = true;
```
Comment (evacchi) [inline kogito-codegen/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java:181]: "web service" could be mistaken for SOAP, `useRestServices` seems better

**Comment by mariofusco** [comment]
Replaced by https://github.com/kiegroup/kogito-runtimes/pull/506

**Pull request closed without merging**